kit/azure/buildingblocks/standard-vnet-configuration/main-backend.tf

```diff
@@ -14,35 +18,63 @@ data "azurerm_subscription" "sta_subscription" {
   subscription_id = local.sta_subscription_id
 }
 
+data "azurerm_storage_account" "tfstates" {
+  name                = local.sta_name
+  resource_group_name = local.sta_rg_name
+}
+
+# resource "azurerm_storage_container" "tfstates" {
+#   name = "tfstates-standard-vnet"
+#   storage_account_name = data.azurerm_storage_account.tfstates.name
+#   container_access_type = "blob"
+# }
+
+resource "azapi_resource" "container" {
+  type      = "Microsoft.Storage/storageAccounts/blobServices/containers@2022-09-01"
+  name      = "tfstates-standard-vnet"
+  parent_id = "${data.azurerm_storage_account.tfstates.id}/blobServices/default"
+  body = jsonencode({
+    properties = {
+      defaultEncryptionScope      = "$account-encryption-key"
```

Comment:
Q: Which encryption scope does `$account-encryption-key` map to? Is it the one from the account and if yes, do we need to set it explicitly?

What's the advantage of a dedicate scope for  using the scope created for the storage account?

Reply:
this is the default one that the other containers have, in general you can change the encryption key on the blob level if e.g. different teams use a same storage account. can be microsoft managed or customer managed keys

Reply:
If it's the default one, can we leave it out of the definition or does it have to be here?

Reply:
it has to be there, azapi is a bit different, needs more manual intervention